Sources/Git/Commit.swift

```diff
@@ -0,0 +1,136 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2018 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import clibgit
+
+/// The commit object.
+public final class Commit {
+    typealias Handle = OpaquePointer
+
+    /// The commit struct pointer.
+    let handle: OpaquePointer
+
+    /// The object ID of the commit.
+    public lazy var objectID: ObjectID = {
+        return ObjectID(git_commit_id(handle)!.pointee)
+    }()
+
+    /// The committer of the commit (user who commited).
+    public lazy var committer: Signature? = {
+        return git_commit_committer(handle).map(Signature.init(pointer:))
+    }()
+
+    /// The author of the commit (user who wrote the code).
+    public lazy var author: Signature? = {
+        return git_commit_author(handle).map(Signature.init(pointer:))
+    }()
+
+    /// The number of parents of the commit.
+    public lazy var parentCount: Int = {
+        return Int(git_commit_parentcount(handle))
+    }()
+
+    /// The full message of the commit.
+    public lazy var message: String = {
+        return String(cString: git_commit_message(handle))
+    }()
+
+    init(handle: Handle, objectID: ObjectID? = nil) {
+        self.handle = handle
+
+        if let objectID = objectID {
+            self.objectID = objectID
+        }
+    }
+
+    // Get parent commit, or nil if this is the first commit.
+    public func getParent() throws -> Commit? {
+        guard parentCount > 0 else { return nil }
+
+        var parentHandle: Handle?
+        try wrap { git_commit_parent(&parentHandle, self.handle, 0) }
+        return Commit(handle: parentHandle!)
+    }
+
+    // Get commit tree.
+    public func getTree() throws -> Tree {
+        var treeHandle: Tree.Handle?
+        try wrap { git_commit_tree(&treeHandle, handle) }
+        return Tree(handle: treeHandle!)
+    }
+
+    deinit {
+        git_commit_free(handle)
+    }
+}
+
+extension Commit: Hashable {
+    public static func == (lhs: Commit, rhs: Commit) -> Bool {
+        return lhs.objectID == rhs.objectID
+    }
+
+    public var hashValue: Int {
+        return objectID.hashValue
+    }
+}
+
+/// An action signature (e.g. for committers, taggers, etc)
+public struct Signature {
+    /// Pointer to the git signature struct (doesn't need to be freed).
```

Comment:
This seems unlikely to be completely true... presumably it only doesn't need to be freed as long as some other object it is inside doesn't need to be freed? But as written, I don't think that is guaranteed? We might need this object to hold a reference to that object to make this safe.